src/Generators/Microsoft.Gen.Logging/Parsing/SymbolLoader.cs

```diff
@@ -52,16 +53,16 @@ internal static class SymbolLoader
         Compilation compilation,
         Action<DiagnosticDescriptor, Location?, object?[]?> diagCallback)
     {
-        var loggerSymbol = compilation.GetTypeByMetadataName(ILoggerType);
-        var logLevelSymbol = compilation.GetTypeByMetadataName(LogLevelType);
-        var loggerMessageAttributeSymbol = compilation.GetTypeByMetadataName(LoggerMessageAttribute);
-        var logPropertiesAttributeSymbol = compilation.GetTypeByMetadataName(LogPropertiesAttribute);
-        var tagProviderAttributeSymbol = compilation.GetTypeByMetadataName(TagProviderAttribute);
-        var tagNameAttributeSymbol = compilation.GetTypeByMetadataName(TagNameAttribute);
-        var tagCollectorSymbol = compilation.GetTypeByMetadataName(ITagCollectorType);
-        var logPropertyIgnoreAttributeSymbol = compilation.GetTypeByMetadataName(LogPropertyIgnoreAttribute);
-        var dataClassificationAttribute = compilation.GetTypeByMetadataName(DataClassificationAttribute);
-        var noDataClassificationAttribute = compilation.GetTypeByMetadataName(NoDataClassificationAttribute);
+        var loggerSymbol = compilation.GetBestTypeByMetadataName(ILoggerType);
```

Comment:
This aligns with the symbol lookup behavior from dotnet runtime logging generator.